es6-shim.js

```diff
@@ -337,8 +349,9 @@
         // special case support for `arguments`
         return new ArrayIterator(o, 'value');
       }
-      var itFn = o[$iterator$];
+      var itFn = ES.GetMethod(o, $iterator$);
       if (!ES.IsCallable(itFn)) {
```

Comment:
GetMethod doesn't complain if the property is missing, just if it's present but not callable.  As the comment inside that check indicates, the explicit check gives us better error messages in case the property is missing or null (in which case ES.GetMethod returns null or undefined).

It's not a correctness issue, since (as in the spec) the `_call` immediately following will throw the appropriate TypeError if `itFn` is null or undefined.  But you'd be looking at a pretty opaque TypeError in that case, just complaining that "undefined is not a function" or something like that.  This test lets us give a more friendly type error.